/**
Start writing the server by defining a service.
**/
syntax = "proto3";

/** Importing the definition of Timestamp
    from the path as defined by Google.**/
import "google/protobuf/timestamp.proto";

message Location {
    double lat = 1;
    double lng = 2;
}

// Enumeration type for RideType
enum RideType {
    REGULAR = 0;
    POOL = 1;
}

message StartRequest {
    int64 car_id = 1;
    string driver_id = 2;
    repeated string passenger_ids = 3;
    RideType type = 4;
    Location location = 5;
    google.protobuf.Timestamp time = 6;
}

// Format interface of the response
message StartResponse {
    string id = 1;
}

// Define service Rides with a method call Start
service Rides {
    rpc Start(StartRequest) returns (StartResponse) {}
}